compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2035,21 +2038,74 @@ public void visit(BLangRecordLiteral recordLiteral) {
-                    if (Symbols.isOptional(bField.symbol)) {
-                        continue;
+                BRecordType spreadExprRecordType = (BRecordType) spreadOpExprType;
+                if (!spreadExprRecordType.sealed) {
+                    // More than one inclusive-type-descriptors with spread-fields are not allowed.
+                    if (noOfInclusiveTypes > 0) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MULTIPLE_INCLUSIVE_TYPES,
+                                recordLiteral.expectedType.getKind().typeName());
+                    } else {
+                        noOfInclusiveTypes = noOfInclusiveTypes + 1;
                     }
+                }
 
+                LinkedHashMap<String, BField> fieldsInRecordLiteral = ((BRecordType) spreadOpExpr.type).fields;
+                boolean isSpreadExprRecordTypeSealed = spreadExprRecordType.sealed;
+                for (Object fieldName : names) {
+                    if (fieldsInRecordLiteral.containsKey(fieldName)) {
+                        if (fieldsInRecordLiteral.get(fieldName).type.tag != TypeTags.NEVER) {
+                            // An error is logged if the same field is occurred, and it is not defined as a never-type
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL_SPREAD_OP,
+                                    recordLiteral.expectedType.getKind().typeName(), fieldName, spreadOpField);
+                        }
+                    } else {
+                        if (!isSpreadExprRecordTypeSealed) {
+                            // Log an error if the spread-field has a inclusive type
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.INCLUSIVE_RECORD_TYPE_IN_SPREAD_FIELD,
+                                    spreadOpExpr);
+                            break;
+                        }
+                    }
+                }
+
+                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
                     String name = bField.name.value;
-                    if (names.contains(name)) {
-                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL_SPREAD_OP,
-                                        recordLiteral.expectedType.getKind().typeName(), name, spreadOpField);
+                    if (!names.contains(name)) {
+                        if (bField.type.tag == TypeTags.NEVER) {
+                            if (!neverTypedKeys.contains(name)) {
+                                neverTypedKeys.add(name);
+                            }
+                        } else {
+                            if (neverTypedKeys.contains(name)) {
+                                neverTypedKeys.remove(name);
+                            } else {
+                                if (noOfInclusiveTypes > 0 && spreadExprRecordType.sealed) {
```

Comment:
Why do we need to check `spreadExprRecordType.sealed`? Is it to avoid multiple errors?

Reply:
Yeah.